modules/local/prelude/main.nf

```diff
@@ -0,0 +1,335 @@
+
+    if (pixel_datatype == null || pixel_datatype.toSet().size() != 1 ||
+        !(pixel_datatype.toSet()[0] ==~ /[u]?int(8|16|32)|float|double/)  // There are more bit|complex|double-complex
+        ) {
+        output_xml.add(
+            ["Type", pixel_datatype.toString(), "Consistent valid datatypes (uint8, float16...)", cross]
+        )
+    }
+    else {
+        output_xml.add(
+            ["Type", pixel_datatype.toString(), "Consistent valid datatypes (uint8, float16...)", check]
+        )
+    }
+
+    exposure_time = xml.'**'.findAll {
+            node -> node.name() == 'Plane'
+        }
+        .collect {
+            node ->
+                [
+                    node.@ExposureTime.toDouble(),
+                    node.@ExposureTimeUnit.toString()
+                ]
+        }
+        .toSet()
+
+    if (exposure_time == null || exposure_time.size() != 1 || exposure_time[0] == null || exposure_time[1] == null || exposure_time[1] == "") {
+        output_xml.add(
+            ["ExposureTime|ExposureTimeUnit", exposure_time.toString(),
+            "Consistent valid exposure time and units", cross]
+        )
+    }
+    else {
+        output_xml.add(
+            ["ExposureTime|ExposureTimeUnit", exposure_time.toString(),
+            "Consistent valid exposure time and units", check]
+        )
+    }
+
+    output_file_xml = prefix + "_${meta.id}_xml_mqc.tsv"
+    def f1          = task.workDir.resolve(output_file_xml)
+    f1.text         = output_xml*.join("\t").join("\n")
+}
+
+process SUMMARY_MARKERSHEET_LITERAL {
+    tag "$meta.id"
+    label 'process_single'
+
+    input:
+    val(meta)
+    val(markersheet)
+
+    output:
+    path output_file_markersheet, emit: output
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    exec:
+    def args        = task.ext.args ?: ''
+    def prefix      = task.ext.prefix ?: "summary"
+
+    output_file_markersheet = prefix + "_${meta.id}_markersheet_mqc.tsv"
+    //def f1                  = task.workDir.resolve(output_file_samplesheet)
+    //f1.text                 = meta.map {
+    output = [[
+            "channel_number",
+            "cycle_number",
+            "excitation_wavelength",
+            "emission_wavelength",
+            "exposure_time",
+            "marker_name",
+            "filter",
+            "exposure",
+            "background",
+            "remove",
+            "exposure_time_unit"]]
+
+    markersheet.each {
+        output.add([
+                        it.channel_number,
+                        it.cycle_number,
+                        it.excitation_wavelength,
+                        it.emission_wavelength,
+                        it.exposure_time,
+                        it.marker_name,
+                        it.filter,
+                        it.exposure,
+                        it.background,
+                        it.remove,
+                        it.exposure_time_unit
+                    ])
+            }
```

Comment:
Having the header row and content row ordering separated seems like an opportunity for bugs if they're not kept in sync. Could you define the field values as a list of strings first, then append that into `output`, then use that same list to index into `it` in the `each`? That way there would be nothing to sync.